### PR TITLE
Adds ActiveDurationText to health-composables

### PR DESCRIPTION
**Pull request by garanj**
#### WHAT

Adds the first Composable to `health-composables`. This is the `ActiveDurationText` composable to make it easier to show a chronometer in an exercise.

#### WHY

When using `ExerciseClient` from Health Services, creating a chronometer that ticks correctly can be time consuming.

#### HOW

Creates a Composable that takes values from `ExerciseUpdate` to display a chronometer for active duration.

#### Checklist :clipboard:
- [x] Add explicit visibility modifier and explicit return types for public declarations
- [x] Run spotless check
- [x] Run tests
- [x] Update metalava's signature text files

### CODE REVIEW

File: health-composables/src/main/java/com/google/android/horologist/health/composables/ActiveDurationText.kt [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+            val hours = activeSeconds / 3600
+            val minutes = (activeSeconds % 3600) / 60
+            val seconds = activeSeconds % 60
+            formatTemplate.format(hours, minutes, seconds)
+        }
+    }
+
+    // When the component is not visible, then the ticker should stop, and only resume when the UI
+    // component becomes visible again.
+    DisposableEffect(lifecycleOwner) {
```
Comment (yschimke) [inline health-composables/src/main/java/com/google/android/horologist/health/composables/ActiveDurationText.kt:101]: I'm happy to land this, but I think you can use `lifecycleOwner.lifecycle.repeatOnLifecycle(Lifecycle.State.RESUMED)` for this

File: health-composables/src/main/java/com/google/android/horologist/health/composables/ActiveDurationText.kt [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+ * @param lifecycleOwner lifecycle owner.
+ */
+
+@ExperimentalHorologistHealthComposablesApi
+@Composable
+public fun ActiveDurationText(
+    checkpoint: ExerciseUpdate.ActiveDurationCheckpoint,
+    state: ExerciseState,
+    modifier: Modifier = Modifier,
+    formatTemplate: String = ActiveDurationDefaults.HH_MM_SS,
```
Comment (yschimke) [inline health-composables/src/main/java/com/google/android/horologist/health/composables/ActiveDurationText.kt:75]: I think we can have a cleaner separation here.  Perhaps make Text a Slot, but given the value on these important boundaries?

**Review by yschimke — APPROVED** [review]
I'm keen to land this as a starting point for health composables, but will follow up with some PRs for discussion on improving the approach.

File: health-composables/src/main/java/com/google/android/horologist/health/composables/ActiveDurationText.kt (hunk 2 above)
Comment (garanj) [reply health-composables/src/main/java/com/google/android/horologist/health/composables/ActiveDurationText.kt:75]: Sounds good, so like `slot: (formattedDuration) -> Unit` ?

File: health-composables/src/main/java/com/google/android/horologist/health/composables/ActiveDurationText.kt (hunk 1 above)
Comment (garanj) [reply health-composables/src/main/java/com/google/android/horologist/health/composables/ActiveDurationText.kt:101]: Ok cool, I need to dig into this better to understand this.

File: health-composables/src/main/java/com/google/android/horologist/health/composables/ActiveDurationText.kt [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+private fun nextTimeForOffset(timeMillis: Long, offsetMillis: Long) =
+    ((timeMillis - offsetMillis) / 1000) * 1000 + 1000 + offsetMillis
+
+/**
+ * Calculates the active duration, taking into account any elapsed time since the
+ * [ExerciseUpdate.ActiveDurationCheckpoint] was delivered from Health Services.
+ *
+ * @param checkpoint The checkpoint from the most recent state change.
+ * @param state The current state of the exercise.
+ * @return The current active duration, in seconds since epoch.
```
Comment (ithinkihaveacat) [inline health-composables/src/main/java/com/google/android/horologist/health/composables/ActiveDurationText.kt:184]: nit: Can you return an actual java.time.Duration? Somewhat useful to retain/propagate the type information and the javadoc keeps talking about duration anyway.

**Pull request merged**